File: src/0150-evaluate-reverse-polish-notation/solution.cpp
class Solution {
private: 
    long long int pop(stack<long long int>& s) {
        long long int t = s.top();
        s.pop();
        return t;
    }
public:
    int evalRPN(vector<string>& t) {
        stack<long long int> s;
        for(string c : t) {
            if (c == "+") s.push(pop(s) + pop(s));
            else if (c == "-") s.push(-pop(s) + pop(s));
            else if (c == "*") s.push(pop(s) * pop(s));
            else if (c == "/") s.push( (double)1 / pop(s) * pop(s));
            else s.push(stoi(c));
        }
        return pop(s);
    }
};

File: src/0150-evaluate-reverse-polish-notation/solution.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for t in tokens:
            if t == "+": stack.append(stack.pop() + stack.pop())
            elif t == "-": stack.append(-stack.pop() + stack.pop())
            elif t == "*": stack.append(stack.pop() * stack.pop())
            elif t == "/": stack.append(int(1 / (stack.pop()) * stack.pop()))
            else: stack.append(int(t))
        return stack[0]

